### PR TITLE
rework the arcatraz intro script, using spawn_group and worldstates

### PR DESCRIPTION
**Pull request by miraco**
## 🍰 Pullrequest

This will let the Intro be like on retail. Its a pretty simple event:

Arcatraz Defender Group will handle the respawn of Protean Nightmare. 
Arcatraz Warden Groups will handle the respawn of Protean Horrors 

### Proof

- None

### Issues

- None

### How2Test

- None

### Todo / Checklist

- [X] None

### CODE REVIEW

File: src/game/AI/ScriptDevAI/scripts/outland/tempest_keep/arcatraz/arcatraz.cpp [hunk 1]
```diff
@@ -90,30 +90,20 @@ static const DialogueEntry aArcatrazDialogue[] =
...
+    m_resetDelayTimer(0)
 {
     Initialize();
 }
 
 void instance_arcatraz::Initialize()
 {
+    instance->GetVariableManager().SetVariable(WORLD_STATE_PROTEAN_HORROR, 1);
+    instance->GetVariableManager().SetVariable(WORLD_STATE_PROTEAN_NIGHTMARE, 1);
+    m_WardenGroup = false;
```
Comment (killerwife) [inline src/game/AI/ScriptDevAI/scripts/outland/tempest_keep/arcatraz/arcatraz.cpp:102]: you prolly wanna do m_wardenGroup(false) above in the constructor instead

Comment (miraco) [reply src/game/AI/ScriptDevAI/scripts/outland/tempest_keep/arcatraz/arcatraz.cpp:102]: done

File: src/game/AI/ScriptDevAI/scripts/outland/tempest_keep/arcatraz/arcatraz.cpp [hunk 2]
```diff
@@ -172,11 +162,31 @@ void instance_arcatraz::OnCreatureCreate(Creature* creature)
...
+// When Arcatraz Defender spawn_group is dead, Protean Nightmare stop spawning
+void instance_arcatraz::OnCreatureGroupDespawn(CreatureGroup* pGroup, Creature* /*pCreature*/)
+{
+    if (pGroup->GetGroupId() == SPAWN_GROUP_WARDEN_01)
+        if (m_WardenGroup)
+            instance->GetVariableManager().SetVariable(WORLD_STATE_PROTEAN_HORROR, 0);
+        else
+            m_WardenGroup = true;
+
+    if (pGroup->GetGroupId() == SPAWN_GROUP_WARDEN_02)
```
[the remark this answers is not in the document]
Comment (miraco) [reply src/game/AI/ScriptDevAI/scripts/outland/tempest_keep/arcatraz/arcatraz.cpp:176]: done

[the remark this answers is not in the document]
Comment (killerwife) [reply src/game/AI/ScriptDevAI/scripts/outland/tempest_keep/arcatraz/arcatraz.cpp:176]: Push the commit then boss :D

[the remark this answers is not in the document]
Comment (miraco) [reply src/game/AI/ScriptDevAI/scripts/outland/tempest_keep/arcatraz/arcatraz.cpp:177]: oh my bad :D

[the remark this answers is not in the document]
Comment (miraco) [reply src/game/AI/ScriptDevAI/scripts/outland/tempest_keep/arcatraz/arcatraz.cpp:177]: I added another change. The Protean Horror/Protean Nightmare shouldnt give any loot/xp/rep

File: src/game/AI/ScriptDevAI/scripts/outland/tempest_keep/arcatraz/arcatraz.cpp [hunk 3]
```diff
@@ -160,6 +160,21 @@ void instance_arcatraz::OnCreatureCreate(Creature* creature)
             break;
     }
 }
+void instance_arcatraz::OnCreatureRespawn(Creature* creature)
+{
+    switch (creature->GetEntry())
+    {
+        case NPC_PROTEAN_NIGHTMARE:
+        case NPC_PROTEAN_HORROR:
+        if (creature->HasStringId(STRING_ID_ENTRANCE_GROUP))
```
Comment (killerwife) [inline src/game/AI/ScriptDevAI/scripts/outland/tempest_keep/arcatraz/arcatraz.cpp:169]: missing indent, if should be one tab further

Comment (miraco) [reply src/game/AI/ScriptDevAI/scripts/outland/tempest_keep/arcatraz/arcatraz.cpp]: done

**Pull request merged**